app/src/main/kotlin/com/wire/android/mapper/MessageContentMapper.kt

```diff
@@ -250,20 +250,31 @@ class MessageContentMapper @Inject constructor(
         }
     }
 
-    fun toText(conversationId: ConversationId, content: MessageContent) = MessageBody(
-        when (content) {
-            is MessageContent.Text -> UIText.DynamicString(content.value, content.mentions)
-            is MessageContent.Unknown -> UIText.StringResource(
-                messageResourceProvider.sentAMessageWithContent, content.typeName ?: "Unknown"
-            )
+    fun toText(conversationId: ConversationId, content: MessageContent): UIMessageContent.TextMessage {
+        val messageTextContent = (content as? MessageContent.Text)
 
-            is MessageContent.FailedDecryption -> UIText.StringResource(R.string.label_message_decryption_failure_message)
-            else -> UIText.StringResource(messageResourceProvider.sentAMessageWithContent, "Unknown")
-        },
-        quotedMessage = (content as? MessageContent.Text)?.quotedMessageDetails?.let { mapQuoteData(conversationId, it) }
-    ).let { messageBody -> UIMessageContent.TextMessage(messageBody = messageBody) }
+        val quotedMessage = messageTextContent?.quotedMessageDetails?.let { mapQuoteData(conversationId, it) }
+            ?: if (messageTextContent?.quotedMessageReference?.quotedMessageId != null) {
+                UIQuotedMessage.UnavailableData
+            } else {
+                null
+            }
+
+        return MessageBody(
+            when (content) {
+                is MessageContent.Text -> UIText.DynamicString(content.value, content.mentions)
+                is MessageContent.Unknown -> UIText.StringResource(
+                    messageResourceProvider.sentAMessageWithContent, content.typeName ?: "Unknown"
```

Comment:
Should this "Unknown" string should be somehow, localized as well ?

Reply:
Done